resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/BasicExecutorFeatureStep.scala

```diff
@@ -45,10 +45,15 @@ private[spark] class BasicExecutorFeatureStep(
   private val blockManagerPort = kubernetesConf
     .sparkConf
     .getInt(BLOCK_MANAGER_PORT.key, DEFAULT_BLOCKMANAGER_PORT)
+  private val hostNetwork = kubernetesConf.get(KUBERNETES_EXECUTOR_HOSTNETWORK)
```

Comment:
`hostNetwork` might be enabled when this is true or executor pod template contains right